Add end to end test coverage and main entry point for bucket

cintamani opened the pull request:
Depends on: https://github.com/DEFRA/defra-ruby-aws/pull/3 

Add the end-to-end feature test and document the Gem's configuration and usage.

irisfaraway commented on README.md:
```diff
@@ -20,9 +20,43 @@ Or install it yourself as:
+    name: "defra-ruby-aws",
+    # AWS bucket access credentials, required
+    credentials: {
+      access_key_id: "ACCES_KEY_ID",
```
```suggestion
      access_key_id: "ACCESS_KEY_ID",
```

Typo :) Are these the names of env vars? If so I think we should include that in the comment.

irisfaraway commented on README.md:
````diff
@@ -20,9 +20,43 @@ Or install it yourself as:
+Add a new bucket with:
+
+```
+# config/initializers/defra-ruby-aws.rb
````
```suggestion
# config/initializers/defra_ruby_aws.rb
```

To follow the usual Rails file naming scheme.

irisfaraway reviewed (commented):
Just focusing on the docs for now – a couple questions and suggestions!

cintamani committed:
Add end to end test coverage and main entry point for bucket

Add the end-to-end feature test and document the Gem's configuration and usage.
Add main entry point to the buckets and move AWS methods in the correct module

cintamani commented on README.md:
````diff
@@ -20,9 +20,43 @@ Or install it yourself as:
+
+```
+# config/initializers/defra-ruby-aws.rb
+require "defra-ruby-aws"
````
```suggestion
require "defra_ruby/aws"
```

cintamani committed:
Apply suggestions from code review

Co-Authored-By: Iris Faraway <iris.faraway@gmail.com>

Cruikshanks reviewed (approved):
LGTM, though as @irisfaraway had a first pass at this might just want to check in with her before merging.
